src/utilities/HiFiMath.ts

```diff
@@ -0,0 +1,593 @@
+     * **Positive yaw** means that the aircraft will rotate **counter-clockwise** when viewing the aircraft from above.
+     * Units here are degrees.
+     */
+    yawDegrees: number;
+    /**
+     * Consider an aircraft: "Roll" is defined as "rotation about the axis running from nose to tail".
+     * **Positive roll** means that the aircraft's **right wing will move closer to the ground**.
+     * **Negative roll** means that the aircraft's **left wing will move closer to the ground**.
+     * Units here are degrees.
+     */
+    rollDegrees: number;
+
+    /**
+     * Construct a new `OrientationEuler3D` object. All parameters are optional. Unset parameters will be set to `0.0`. Remember, all units for member variables are `degrees`.
+     */
+    constructor({ pitchDegrees = 0.0, yawDegrees = 0.0, rollDegrees = 0.0 }: { pitchDegrees?: number, yawDegrees?: number, rollDegrees?: number } = {}) {
+        this.pitchDegrees = sanitizeAngleDegrees(pitchDegrees);
+        this.yawDegrees = sanitizeAngleDegrees(yawDegrees);
+        this.rollDegrees = sanitizeAngleDegrees(rollDegrees);
+    }
+}
+
+/**
+ * Aside from the 3 angles Yaw, Pitch, Roll defining an orientation, euler angles requires 
+ * to define the order in witch the individual yaw, pitch roll rotations are combined.
+ * There are 6 orders possible identified by the HiFiEulerOrder enum.
+ * 
+ *  For example, the order YawPitchRoll is describing the following sequence
+ *  starting from the base 3d frame,
+ *  1/ Yaw, rotating around the vertical axis
+ *  2/ Pitch, rotating around the right axis 
+ *  3/ Roll, rotating around the front axis
+ *  the resulting 3d frame orientation is relative to the base frame.
+ */
+export enum OrientationEuler3DOrder {
+    PitchYawRoll = "PitchYawRoll",
+    YawPitchRoll = "YawPitchRoll",
+    RollPitchYaw = "RollPitchYaw",
+    RollYawPitch = "RollYawPitch",
+    YawRollPitch = "YawRollPitch",
+    PitchRollYaw = "PitchRollYaw",
+}
+
+/**
+ * Compute the orientation quaternion from the specified euler angles.
+ * The resulting quaternion is the rotation transforming from combining the euler angles rotations in the specified order
+ * 
+ * For example, the order YawPitchRoll is computed as follow:
+ *  starting from the base 3d frame,
+ *  1/ Yaw, rotating around the vertical axis
+ *  2/ Pitch, rotating around the right axis 
+ *  3/ Roll, rotating around the front axis
+ *  the resulting 3d frame orientation is relative to the base frame.
+ *  The resulting rotation is defining the 'rotated' space relative to the 'base' space.
+ *  A vector Vr in "rotated' space and its equivalent value Vb in the'base' space is computed as follow:
+ *  Vb = [P][Y][R] Vr
+ * 
+ * @param euler - The euler angles.
+ * @param order - The euler order convention.
+ * 
+ * @returns The end resulting quaternion defined from the euler angles combination
+ */
+export function eulerToQuaternion(euler: OrientationEuler3D, order: OrientationEuler3DOrder): Quaternion {
```

Comment:
This, and eulerFromQuaternion below, only apply for a certain axis configuration. Maybe that should be documented.